### DISCUSSION TITLE
[Feature] Auto-backup only on wifi

### DISCUSSION THREAD

**Discussion by ghost** [immich-app/immich#5096] [Feature Request]
### The feature

Setting that makes it so that background backups will only occur when connected to a wifi connection (i.e. not a cell network).

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by schuhbacca** [comment]
![Screenshot_20231117-070642.png](https://github.com/immich-app/immich/assets/5739463/999429dd-61d9-43fa-9c02-7a2ea5b1d7ca)

**Reply by ghost** [reply]
This feature is not on the iOS version

**Reply by chk86** [reply]
Yeah, that toggle is android only. I asked about this in the discord and learned that in iOS you have to use the iOS settings app to turn off cellular data and/or background refresh. No slick looking toggle for apple. 😩

**Reply by alextran1502** [reply]
In the Background App Refresh setting in the phone, you can set it to use Wifi only

**Reply by chk86** [reply]
Yeah, the specific setting is "Cellular Data" and you can just toggle it off. So Settings>Immich>Cellular Data.

**Reply by chk86** [reply]
But that also means you can't use the app when you're not on wifi. At least it doesn't work as well.

**Reply by chk86** [reply]
If you turn it to Wifi only in Background App Refresh, I don't think that would prevent the app from uploading though, @alextran1502 . Like if you open the app and have foreground backup on, and you're on cellular data, it would totally upload even over cell data, right?

**Reply by alextran1502** [reply]
@chk86 Yeah that is only for background tasks. It doesn't relate to how you access and backup in the foreground

**Reply by ghost** [reply]
The thing is, I don’t want to set my entire background app refresh to wifi only. I only want Immich to do so which is not possible from the settings app.

**Reply by Batwam** [reply]
People seem to be suggesting workarounds (turn off mobile data, etc...) rather than reasons why this is possible on Android but not iOS version. 

Is this because this is a limitation of Android? gPhotos seems to allow disabling uploads on mobile data.

![image](https://github.com/immich-app/immich/assets/56710655/b92126b4-92f5-4ac1-82d5-97bc1c442916)

**Comment by tdashmike** [comment]
I would like this feature as well so that it won't upload on cellular. Changing it to Wi-Fi in the iOS system settings disables cellular for all apps.

**Comment by leepfrog-ger** [comment]
It would be great to have that added to the iOS app.
Without knowing the details I assume that it is technically possible on iOS because other apps have that feature. e.g. the Synology app "Photos Mobile" has exactly that toggle: restrict background backups to wifi only.

I know that I can restrict it to charging only, however there are several cases where that would not be enough, for example when charging the phone when not at home (hotel, or while driving).
Also, when being on cellular the "foreground" sync is starting as soon as I open the app. 
Think of not being at home, having shot a few larger videos and then opening Immich to look through other pictures from the app. With 5G this can really quickly eat through your data plan and/or battery.

I don't want to disable background refresh or cellular data for immich completely from the iOS settings as I still want to be able to use the app on cellular network, I just want to make sure battery/data is conserved while being on the move.

**Comment by richardacre** [comment]
Any news on this? I'm eager to move from Google Photos to self-hosted Immich and had actually assumed I could set backups to WiFi-only while still being able to browse search etc over mobile data

**Reply by Batwam** [reply]
> set backups to WiFi-only

There is an option for that now, at least for iOS. Just noticed it recently.
![image](https://github.com/user-attachments/assets/0afd6c87-293c-4e13-b05e-b48ab46433fd)
